### ISSUE TITLE
Syntax Highlighting

### ISSUE THREAD

**Issue by davej** [breuleux/earl-grey#6]
Is there any syntax highlighting definitions available? Or is there a language that earl grey shares similar syntax with, so I can piggy-back on the syntax highlighting defined for that language?

**Comment by breuleux** [comment]
I made an `earl-mode` for emacs [here](https://github.com/breuleux/earl-grey-mode). I also have a CodeMirror module [here](https://github.com/breuleux/repple/blob/master/src/earlmode.eg) which is currently embedded in repple, the engine behind [this page](http://breuleux.github.io/earl-grey/repl/) (I'll try to find the time to make a pull request to include it in CodeMirror, but I'd probably need to rewrite it in JS to get it accepted). Then I also have a mode for highlight.js [here](https://github.com/breuleux/eg-doc/blob/master/hl.eg). So, yeah, there are some definitions around, but it's kind of messy and scattered.

EG is a bit tricky to highlight as well since it's not keyword-based: `alice bob: charlie` should highlight alice as a keyword, for example.

I believe Python might work okay as an alternative mode since most control structures have the same syntax.

**Comment by davej** [comment]
Thanks, did a bit of trial and error and python works ok but I found CoffeeScript works best. I quickly added dot-strings and `;;` comments so it works quite well now.

**Comment by breuleux** [comment]
Yeah, now that I think of it, CoffeeScript would work well too. What editor are you using? If you do end up tweaking a decent mode for EG, put it up somewhere and I'll link to it when I end up adding a section on editor support.

**Comment by davej** [comment]
Will do. Using Atom. Highlighting goes a bit horrible when using `dash-er-ized` variable names but I'll post it here if I get it working well.

**Comment by breuleux** [comment]
For those who are interested, I have written a guide on syntax highlighting for Earl Grey:

http://breuleux.github.io/earl-grey/contrib.html

Also, someone is developing an Atom mode here: https://github.com/MadcapJake/language-earl-grey. Probably best not to make more than one :)
